luma_core/src/vi.rs

```diff
@@ -37,23 +37,66 @@ impl Xfb {
         self.height
     }
 
-    /// Get the stride of this XFB, given the YUYV format this is always width × 2.
-    pub fn stride(&self) -> usize {
-        // YUYV always takes two bytes per pixel.
-        self.width * 2
+    /// Get the stride of this XFB.  This is always equal to width for now.
+    pub fn stride_in_u16(&self) -> usize {
+        self.width
+    }
+
+    /// Get the stride of this XFB in bytes.  Given the YUYV format this is always equal to
+    /// stride_in_u16() × 2.
+    pub fn stride_in_u8(&self) -> usize {
+        self.stride_in_u16() * 2
+    }
+
+    /// Get a mutable iterator over the rows of this XFB.
+    pub fn iter_mut(&mut self) -> XfbIterMut {
+        XfbIterMut {
+            xfb: self,
+            cur_row: 0,
+        }
     }
 
     /// Return the raw pointer to this XFB.
-    pub fn as_ptr(&self) -> *const u8 {
+    pub fn as_ptr(&self) -> *const u16 {
         self.data.as_ptr()
     }
 
     /// Return the raw mutable pointer to this XFB.
-    pub fn as_mut_ptr(&mut self) -> *mut u8 {
+    pub fn as_mut_ptr(&mut self) -> *mut u16 {
         self.data.as_mut_ptr()
     }
 }
 
+pub struct XfbIterMut<'a> {
+    xfb: &'a mut Xfb,
+    cur_row: usize,
+}
+
+impl<'a> Iterator for XfbIterMut<'a> {
+    type Item = &'a mut [u16];
+
+    fn next(&mut self) -> Option<&'a mut [u16]> {
```

Comment:
would `self.xfb.data.chunks_mut(self.xfb.stride_in_u16()).next()` work here?

Reply:
I’ve ended up replacing the whole `XfbIterMut` struct with `ChunksMut`, which already does the exact same and more.